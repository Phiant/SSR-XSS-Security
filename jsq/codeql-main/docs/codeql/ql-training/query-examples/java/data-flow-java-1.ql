import java

class StringConcat extends AddExpr {
  StringConcat() { getType() instanceof TypeString }
}

from MethodCall ma
where
  ma.getMethod().getName().matches("sparql%Query") and
  ma.getArgument(0) instanceof StringConcat
select ma, "SPARQL query vulnerable to injection."
