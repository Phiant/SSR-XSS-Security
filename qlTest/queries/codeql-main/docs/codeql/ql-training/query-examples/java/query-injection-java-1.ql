import java

from Method m, MethodCall ma
where
  m.getName().matches("sparql%Query") and
  ma.getMethod() = m
select ma, m
