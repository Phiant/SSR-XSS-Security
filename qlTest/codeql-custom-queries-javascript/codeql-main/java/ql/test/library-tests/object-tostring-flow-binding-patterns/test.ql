import java
import semmle.code.java.dispatch.VirtualDispatch

from Call c, Callable c2
where c2 = viableCallable(c)
select c, c2.getQualifiedName()
