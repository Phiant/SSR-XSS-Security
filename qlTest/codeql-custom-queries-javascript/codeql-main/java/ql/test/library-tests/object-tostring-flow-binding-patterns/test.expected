| Test.java:1:14:1:17 | super(...) | java.lang.Object.Object |
| Test.java:4:16:4:23 | super(...) | java.lang.Object.Object |
| Test.java:5:16:5:26 | super(...) | java.lang.Object.Object |
| Test.java:9:21:9:37 | new Alternative(...) | Test$Alternative.Alternative |
| Test.java:10:12:10:31 | test(...) | Test.test |
| Test.java:10:17:10:30 | new Specific(...) | Test$Specific.Specific |
| Test.java:18:14:18:26 | toString(...) | Test$Specific.toString |
| Test.java:22:27:22:39 | hashCode(...) | java.lang.Object.hashCode |
| Test.java:22:57:22:69 | toString(...) | Test$Specific.toString |
